PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-resource-connector-lib-java/src/main/java/com/microsoft/azure/toolkit/intellij/connector/code/ResourceConnectionLineMarkerInfo.java

```diff
@@ -46,12 +46,15 @@ public ResourceConnectionLineMarkerInfo(@Nonnull Connection<?, ?> connection, fi
         this.connection = connection;
     }
 
-    private static Icon getIcon(final AzureServiceResource<?> resource) {
-        final String definitionIconPath = StringUtils.firstNonBlank(resource.getDefinition().getIcon(), AzureIcons.Connector.CONNECT.getIconPath());
-        final AzureIcon icon = Azure.az(AzureAccount.class).isLoggedIn() ?
-                DEFAULT_AZURE_RESOURCE_ICON_PROVIDER.getIcon(resource.getData()) :
-                AzureIcon.builder().iconPath(definitionIconPath).build();
-        return IntelliJAzureIcons.getIcon(icon);
+    private static Icon getIcon(@Nonnull final AzureServiceResource<?> resource) {
+        final AzureIcon resourceIcon = Azure.az(AzureAccount.class).isLoggedIn() ?
+                DEFAULT_AZURE_RESOURCE_ICON_PROVIDER.getIcon(resource.getData()) : null;
+        final AzureIcon definitionIcon = AzureIcon.builder().iconPath(resource.getDefinition().getIcon()).build();
+        return Stream.of(resourceIcon, definitionIcon, AzureIcons.Connector.CONNECT)
+                .filter(Objects::nonNull)
+                .filter(i -> !Objects.equals(AzureIcons.Common.REFRESH_ICON, i))
+                .map(IntelliJAzureIcons::getIcon)
+                .findFirst().orElse(AllIcons.Providers.Azure);
```

Comment:
this is not the correct Azure icon.

Reply:
This is the last fallback in case IntelliJ could not load icon from toolkits.